src/app/store/collection/reducers/items.ts

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable @typescript-eslint/ban-ts-ignore */
+import * as R from 'ramda';
+
+export interface Item {}
+
+export interface ItemsAction<I extends Item> {
+  type: string;
+  payload: I[];
+}
+
+export type ItemsState<I extends Item> = I[];
+
+export const initialState = [];
+
+export default <I extends Item>(SUCCESS: string, itemIdentifier: string) => (
+  state: ItemsState<I> = initialState,
+  action: ItemsAction<I>
+): ItemsState<I> => {
+  switch (action.type) {
+    case SUCCESS: {
+      return R.uniqWith(
+        // @ts-ignore
```

Comment:
Solution was to ensure that 'Item' contains the 'itemIdentifier'. This works for me :

```ts
import * as R from 'ramda';

export type Item<Id extends string = 'id'> = Record<Id, unknown>;

export interface ItemsAction<I extends Item<Id>, Id extends string> {
  type: string;
  payload: I[];
}

export type ItemsState<I extends Item<Id>, Id extends string> = I[];

export const initialState = [];

export default <I extends Item<Id>, Id extends string>(
  SUCCESS: string,
  itemId: string
) => (
  state: ItemsState<I, Id> = initialState,
  action: ItemsAction<I, Id>
): ItemsState<I, Id> => {
  switch (action.type) {
    case SUCCESS: {
      return R.uniqWith(R.eqProps(itemId), R.concat(state, action.payload));
    }
    default:
      return state;
  }
};
```

Reply:
@lutangar I’ve applied my own solution but I’ve reverted it because it breaks other parts :-(

TypeScript is correct yelling at us :
- We define `ItemsState` as `<I extends Item> = I[];`, `Item` being a `{}` interface
- We then use this property in the reducer to be able to `R.eqProps(itemIdentifier),`
- But in `src/app/profiles/store/reducers/subscriptions.reducer.ts` we use `ItemsState<Subscription>`, `Subscription` being a `number`
- It only works because we don’t use the default `item` reducer from `collection` helper, but a custom one. 

But it’s still wrong as here `Subscription` **is not** a valid `Item` … and we may create a bug by confusing a `number` with an `object` with `id` …

I create an issue from this in order to merge it in current state